src/components/datagrid/data_grid_types.ts

```diff
@@ -268,6 +268,19 @@ export type EuiDataGridProps = OneOf<
   'aria-label' | 'aria-labelledby'
 >;
 
+export interface EuiDataGridRefProps {
+  /**
+   * Allows manually focusing the specified cell in the grid.
+   *
+   * Using this method is an accessibility requirement if your EuiDataGrid
+   * toggles a modal or flyout - focus must be restored to the grid on close
+   * to prevent keyboard or screen reader users from being stranded.
+   */
+  setFocusedCell({ rowIndex, colIndex }: EuiDataGridCellLocation): void;
```

Comment:
Small suggestion, which de-duplicates the object definition

```suggestion
  setFocusedCell(targetCell: EuiDataGridCellLocation): void;
```